bluemira/radiation_transport/advective_transport.py

```diff
@@ -322,8 +322,18 @@ def _analyse_DN(self, first_wall):
         self._make_flux_surfaces_ib()
 
         # Find the intersections of the flux surfaces with the first wall
-        for flux_surface in self.flux_surfaces:
-            flux_surface.clip(first_wall)
+        for no, flux_surface in enumerate(self.flux_surfaces):
+            try:
+                flux_surface.clip(first_wall)
+            except ValueError:
+                bluemira_warn(
+                    f"No intersection detected betwenn wall and flux_surface {no}"
+                )
+                if no > len(self.flux_surfaces_ib_lfs) and no < len(
+                    self.flux_surfaces_ob_lfs
+                ):
+                    self.flux_surfaces_ob_lfs.pop(no)
+                    self.flux_surfaces_ob_hfs.pop(no)
```

Comment:
Is this fix working for the case of a non intersecting ob flux line ?

Reply:
Here lfs and hfs are indicating (not correctly) lower divertor and higher divertor. So, we are still at the low field side, aka outboard, aka right, and I am popping both lower and upper part of the last open flux surface. The one that doesn't find intersections